lib/db/DB.ts

```diff
@@ -166,13 +166,20 @@ class DB {
       SwapDeal.sync(),
     ]);
 
-    if (shouldInitDb) {
+    if (initDb) {
       // initialize database with the seed nodes for the configured network
       const nodes = defaultNodes(network);
       if (nodes) {
-        await Node.bulkCreate(nodes);
+        const existingNodes = await Models.Node(this.sequelize).findAll();
+        const newNodes = nodes.filter(node => (!existingNodes.find(n => (n.nodePubKey === node.nodePubKey))));
+
+        if (newNodes) {
```

Comment:
Arrays (even when empty) are truthy so this will always resolve to true, even when there are no nodes. Better to check `newNodes.length > 0`.

Reply:
True, sorry to overlook that, thanks!